### PR TITLE
Add support for max_backoff_delay_seconds

### PR DESCRIPTION
**Pull request by amamd07**
In some cases we need to set upper limit to the delta growth. A member `max_backoff_delay_seconds` is added to control the delta growth and force saturation

### CODE REVIEW

File: pyworker/job.py [hunk 1]
```diff
@@ -20,14 +20,15 @@ class Job(object, metaclass=Meta):
     def __init__(self, class_name, database, logger,
                  job_id, queue, run_at, attempts=0, max_attempts=1,
                  attributes=None, abstract=False, extra_fields=None,
-                 reporter=None):
+                 reporter=None, max_delta=-1):
```
Comment (hammady) [inline pyworker/job.py]: I suggest a better representative name: `max_backoff_seconds` or `max_backoff_delay_seconds`.

**Review by hammady — CHANGES_REQUESTED** [review]

**Comment by hammady** [comment]
Please update README.

**Review by hammady — CHANGES_REQUESTED** [review]
Please update README.

**Comment by hammady** [comment]
> Please update README.

By doing that, you will find that the `Worker` class needs to be updated too!
I also think `-1` is not a natural default value while `None` is.

File: README.md [hunk 2]
```diff
@@ -98,6 +98,10 @@ w.sleep_delay = 3
 # maximum attempts before marking the job as permanently failing (default 3)
 w.max_attempts = 5
 
+# sets a cap on how long a worker should wait between retry attempts.
+# if not set, the backoff delay time will grow exponentially (default 60)
```
Comment (hammady) [inline README.md]: To be backward compatible, the default should be unbounded.

File: pyworker/job.py [hunk 3]
```diff
@@ -146,7 +148,7 @@ def set_error_unlock(self, error):
             # set new exponential run_at
             setters.append('run_at = %s')
             delta = (self.attempts**4) + 5
-            if self.max_backoff_delay_seconds > 0 and delta > self.max_backoff_delay_seconds:
+            if self.max_backoff_delay_seconds and delta > self.max_backoff_delay_seconds:
```
Comment (hammady) [inline pyworker/job.py:153]: Can we also set a minimum of 5 in case the user set the max to a very small number (e.g. 1).

**Review by hammady — CHANGES_REQUESTED** [review]

File: pyworker/job.py [hunk 4]
```diff
@@ -28,6 +28,8 @@ def __init__(self, class_name, database, logger,
         self.job_id = job_id
         self.job_name = '%s#run' % class_name
         self.attempts = attempts
+        if max_backoff_delay_seconds:
+            max_backoff_delay_seconds = max(max_backoff_delay_seconds, 2) # max_backoff_delay_seconds can not be less than 2 seconds
```
Comment (hammady) [inline pyworker/job.py]: - Can we make it 5 to match the internal calculation?
- Document this minimum in the README.
- Add a new test to ensure the minimum is honoured.

**Review by hammady — CHANGES_REQUESTED** [review]

**Review by hammady — APPROVED** [review]
